lib/topological_inventory/openshift/collector.rb

```diff
@@ -25,15 +25,24 @@ def initialize(source, openshift_host, openshift_port, openshift_token, default_
     def collect!
       start_collector_threads
 
+      errors = 0
       until finished?
         ensure_collector_threads
 
         notices = []
         notices << queue.pop until queue.empty?
 
-        targeted_refresh(notices) unless notices.empty?
+        begin
```

Comment:
should we cover also full refresh?

Reply:
That's actually already handled by restarting the thread, https://github.com/ManageIQ/topological_inventory-openshift/blob/c16deb93b1b8d1a162eb12c60d9820c480e1dd53/lib/topological_inventory/openshift/collector.rb#L89-L99

Reply:
right, but that should also have sleep? Otherwise it will also get to a kill loop, where it bombards the server

Reply:
This thread is what restarts the collector threads, so it won't get restarted more than every sleep_poll

Reply:
ok :+1: